lib/std/os/linux/s390x.zig

```diff
@@ -0,0 +1,281 @@
+const std = @import("../../std.zig");
+const iovec = std.posix.iovec;
+const iovec_const = std.posix.iovec_const;
+const linux = std.os.linux;
+const SYS = linux.SYS;
+const uid_t = std.os.linux.uid_t;
+const gid_t = std.os.linux.gid_t;
+const pid_t = std.os.linux.pid_t;
+const sockaddr = linux.sockaddr;
+const socklen_t = linux.socklen_t;
+const timespec = std.os.linux.timespec;
+const stack_t = std.os.linux.stack_t;
+const sigset_t = std.os.linux.sigset_t;
+
+pub fn syscall0(number: SYS) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+        : "memory"
+    );
+}
+
+pub fn syscall1(number: SYS, arg1: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+        : "memory"
+    );
+}
+
+pub fn syscall2(number: SYS, arg1: usize, arg2: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+        : "memory"
+    );
+}
+
+pub fn syscall3(number: SYS, arg1: usize, arg2: usize, arg3: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+        : "memory"
+    );
+}
+
+pub fn syscall4(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+        : "memory"
+    );
+}
+
+pub fn syscall5(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+          [arg5] "{r6}" (arg5),
+        : "memory"
+    );
+}
+
+pub fn syscall6(number: SYS, arg1: usize, arg2: usize, arg3: usize, arg4: usize, arg5: usize, arg6: usize) usize {
+    return asm volatile ("svc 0"
+        : [ret] "={r2}" (-> usize),
+        : [number] "{r1}" (@intFromEnum(number)),
+          [arg1] "{r2}" (arg1),
+          [arg2] "{r3}" (arg2),
+          [arg3] "{r4}" (arg3),
+          [arg4] "{r5}" (arg4),
+          [arg5] "{r6}" (arg5),
+          [arg6] "{r7}" (arg6),
+        : "memory"
+    );
+}
+
+pub fn clone() callconv(.Naked) usize {
+    asm volatile (
+        \\.text
+        \\.global __clone
+        \\.hidden __clone
+        \\.type __clone, %function
+        \\__clone:
```

Comment:
```suggestion
```

Reply:
what did you change here?

Reply:
I just removed those lines since they're unnecessary; the advantage of using naked functions over assembly files is that the compiler does this function scaffolding for you but without emitting any prologue/epilogue instructions.

Reply:
github ui was confusing because i'd just woken up. diff looked like it was adding and deleting the same lines